public/app/features/alerting/components/AlertingQueryPreview.tsx

```diff
@@ -49,55 +50,61 @@ export const AlertingQueryPreview: FC<Props> = ({ getInstances, onRunQueries, on
         })}
       </TabsBar>
       <TabContent className={styles.tabContent}>
-        {data && data.state === 'Error' ? (
-          <div className={styles.noQueries}>
-            <h4 className={styles.noQueriesHeader}>There was an error :(</h4>
-            <div>{data.error?.data?.error}</div>
-          </div>
-        ) : queries && queries.length > 0 ? (
-          <AutoSizer style={{ width: '100%', height: '100%' }}>
-            {({ width, height }) => {
-              switch (activeTab) {
-                case Tabs.Instances:
-                  return (
-                    <PreviewInstancesTab
-                      isTested={instances.length > 0}
-                      instances={instances}
-                      styles={styles}
-                      width={width}
-                      height={height}
-                      onTest={onTest}
-                    />
-                  );
-
-                case Tabs.Query:
-                default:
-                  if (data) {
-                    return <PreviewQueryTab data={data} width={width} height={height} />;
-                  }
-                  return (
-                    <div className={styles.noQueries}>
-                      <h4 className={styles.noQueriesHeader}>Run queries to view data.</h4>
-                      <Button onClick={onRunQueries}>Run queries</Button>
-                    </div>
-                  );
-              }
-            }}
-          </AutoSizer>
-        ) : (
-          <div className={styles.noQueries}>
-            <h4 className={styles.noQueriesHeader}>No queries added.</h4>
-            <div>Start adding queries to this alert and a visualisation for your queries will appear here.</div>
-            <div>
-              Learn more about how to create alert definitions <Icon name="external-link-alt" />
-            </div>
-          </div>
-        )}
+        {data &&
+          (data.state === 'Error'
+            ? error(data)
+            : queriesAndInstances(queries, instances, onTest, data, activeTab, onRunQueries))}
       </TabContent>
     </div>
   );
 };
 
+const queriesAndInstances = (
```

Comment:
This should be a proper FC with a capital Name, and pass props to it instead of calling a method.